designer/server/src/test/scala/pl/touk/nussknacker/ui/api/OpenAPIExamplesValidator.scala

```diff
@@ -0,0 +1,109 @@
+package pl.touk.nussknacker.ui.api
+
+import com.networknt.schema.{InputFormat, JsonSchemaFactory, SpecVersion}
+import io.circe.{ACursor, Json}
+import io.circe.yaml.{parser => YamlParser}
+import org.scalactic.anyvals.NonEmptyList
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+
+import scala.jdk.CollectionConverters._
+
+object OpenAPIExamplesValidator {
+
+  def validateExamples(specYaml: String): List[InvalidExample] = {
+    val specJson = YamlParser.parse(specYaml).toOption.get
+    val componentsSchemas =
+      specJson.hcursor
+        .downField("components")
+        .downField("schemas")
+        .focusObjectFields
+        .toMap
+    for {
+      (_, pathItem)  <- specJson.hcursor.downField("paths").focusObjectFields
+      (_, operation) <- pathItem.asObject.map(_.toList).getOrElse(List.empty)
+      invalidExample <- validateRequestExample(operation, componentsSchemas) :::
+        validateResponsesExamples(operation, componentsSchemas)
+    } yield invalidExample
+  }
+
+  private def validateRequestExample(operation: Json, componentsSchemas: Map[String, Json]): List[InvalidExample] =
+    for {
+      (_, mediaType) <- operation.hcursor.downField("requestBody").downField("content").focusObjectFields
+      exampleJson    <- validateMediaTypeExamples(mediaType, operation, isRequest = true, componentsSchemas)
+    } yield exampleJson
+
+  private def validateResponsesExamples(
+      operation: Json,
+      componentsSchemas: Map[String, Json]
+  ): List[InvalidExample] =
+    for {
+      (_, response)  <- operation.hcursor.downField("responses").focusObjectFields
+      (_, mediaType) <- response.hcursor.downField("content").focusObjectFields
+      exampleJson    <- validateMediaTypeExamples(mediaType, operation, isRequest = false, componentsSchemas)
+    } yield exampleJson
+
+  private def validateMediaTypeExamples(
+      mediaType: Json,
+      operation: Json,
+      isRequest: Boolean,
+      componentsSchemas: Map[String, Json]
+  ): List[InvalidExample] = {
+    for {
+      schema <- mediaType.hcursor.downField("schema").focus.toList
+      resolvedSchema = resolveSchemaReferences(schema, componentsSchemas)
+      jsonSchema     = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012).getSchema(resolvedSchema.spaces2)
```

Comment:
why do we use spec version `V202012`? I thought Tapir uses DraftV4 (see `MetaSchema` implementations)

Reply:
When openapi 3.1.0 was published, they point into the alignment with this version (https://www.openapis.org/blog/2021/02/18/openapi-specification-3-1-released) but I can make it configurable and downgrade to draft v4 with notice that we've based on this constants in `MetaSchemaDraft04`

Reply:
let's downgrade